vulture/core.py

```diff
@@ -139,13 +140,13 @@ def __init__(
         message="",
         confidence=DEFAULT_CONFIDENCE,
     ):
-        self.name = name
-        self.typ = typ
-        self.filename = filename
-        self.first_lineno = first_lineno
-        self.last_lineno = last_lineno
-        self.message = message or f"unused {typ} '{name}'"
-        self.confidence = confidence
+        self.name: str = name
```

Comment:
If I change this line to `self.name: int = name`, all tests still pass. Is it possible to add tests that check that all type hints are followed by the code?

Reply:
@jendrikseipp Sorry about that, the issue was that mypy doesn't know the type of the `name` param, so my test incorrectly still passed. I've updated the test to use the `pytype` linter instead, as it does type inference when hints aren't given.
I noticed that running my test with pytest takes just a few seconds, running with tox is extremely slow; not sure why this is.

Reply:
It's easy to figure out if you take a look at the build log 😅
https://github.com/jendrikseipp/vulture/actions/runs/11144462910/job/30971791967?pr=361#step:8:156

Unfortunately not all dependencies have been updated to support Python 3.13 yet 😢
https://github.com/jcrist/msgspec/issues/698
because the contributed fixes have not been merged yet after three months 🤷
https://github.com/jcrist/msgspec/pull/711
You can see that msgspec doesn't support 3.13:
https://pypi.org/project/msgspec/

An advanced build system like Poetry should check Python support automatically
https://python-poetry.org/docs/dependency-specification/
if Vulture would change its backend:
https://python-poetry.org/docs/basic-usage/

Reply:
@reneleonhardt I've updated the PR to only use pytype if python is >= 3.13 (for the time being until it's supported).

Reply:
Hmm, that's unfortunate. I was thinking of switching to `uv`. Would that solve the problem? I won't have time to look into this for the next weeks, though.

Reply:
Nice solution!
I hope msgspec and pytype will support 3.13 soon as it will be released on Monday (but identical to rc3)...